src/warnet/plugins.py

```diff
@@ -0,0 +1,218 @@
+import copy
+import importlib.util
+import inspect
+import os
+import sys
+import tempfile
+from pathlib import Path
+from types import ModuleType
+from typing import Optional
+
+import click
+import inquirer
+import yaml
+from inquirer.themes import GreenPassion
+
+from warnet.constants import (
+    CONTAINER_TAG,
+    ENABLED_TAG,
+    MISSION_TAG,
+    PLUGIN_YAML,
+    PLUGINS_TAG,
+    USER_DIR_TAG,
+    WARNET_USER_DIR_ENV_VAR,
+)
+
+
+class PluginError(Exception):
+    pass
+
+
+imported_modules: dict[str, ModuleType] = {}
+
+
+@click.group(name=PLUGINS_TAG)
+def plugins():
+    """Control plugins"""
+    pass
+
+
+@plugins.command()
+@click.pass_context
+def ls(ctx):
+    """List all available plugins and whether they are activated"""
+    plugin_dir = get_plugins_directory_or(ctx.obj.get(USER_DIR_TAG))
+    if plugin_dir is None:
+        direct_user_to_plugin_directory_and_exit()
+
+    for plugin, status in get_plugins_with_status(plugin_dir):
+        if status:
+            click.secho(f"{plugin.stem:<20} enabled", fg="green")
+        else:
+            click.secho(f"{plugin.stem:<20} disabled", fg="yellow")
+
+
+@plugins.command()
+@click.argument("plugin", type=str, default="")
+@click.pass_context
+def toggle(ctx, plugin: str):
```

Comment:
i dont think this should be a command, but a setting in the `network.yaml` file. Does toggling a plugin in or off automatically deploy it to the cluster or delete an existing plugin pod?

Reply:
Toggling a plugin on makes it visible to Warnet. That way, the user can have, say, 50 plugins in their user folder, but Warnet will only have to parse the contents of a subset of those plugins.

When you say the `network.yaml` file, do you envision this:

```
nodes:
  - name: tank-0001
    config: uacomment=tank0001
    image:
      tag: "26.0"
    addnode:
      - tank-0002
    ln:
      lnd: true
  - name: tank-0002
    config: uacomment=tank0002
    resources:
      limits:
        cpu: 100m
        memory: 128Mi
      requests:
        cpu: 100m
        memory: 128Mi
caddy:
  enabled: true
plugins:
  simln: enabled
  ark-server:
    nodes:
      - name: server-a
      - name: server-b
  my-test-oracle: enabled
  fedimint-server:
      - gateway: tank-0001-ln
```

If so, it is worth noting that I was trying to avoid such a tight coupling. I figured it was simpler to let plugin authors deploy their pods using helm as they saw fit while still providing some loose tie-ins to Warnet such as registering the plugin, specifying a mission value, and specifying a primary container value. That way, plugin users could use `warnet logs` and `warnet status` and see the pods launched by their plugins, and plugin authors could provide users with a way to run the plugin such as `warnet plugins simln launch_activity [activity]` because then plugin users wouldn't need to muck around with some jumble of bash and python script files.

That said, if we want to give plugin authors access to the `network.yaml` file to configure and deploy their plugins, the question becomes how do plugin users know to include that information in the `network.yaml` file? Just thinking out loud, I think we would need something like a `warnet plugins apply [plugin]` which would modify the `network.yaml` file with the appropriate values including SEMVER and maybe domain information to make sure that the `network.yaml` file doesn't drift. Looking at the example I gave above, `my-test-oracle` would need to be `my-test-oracle:mplsgrant:0.0.1` or something like that.

However, if the goal is just to be able to easily share a network with other people, I think sharing a repo in a battle-of-galen-erso manner makes the most sense. The helm deploy actions should work whether they are via the `network.yaml` file or via some bespoke method that the plugin authors come up with.

Reply:
Changed to "plugin consumes warnet"